DCA/adapter.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from PIL import Image
+import argparse
+from pathlib import Path
+
+from adapter_base import AdapterBase
+from utils.tools import *
+from utils.my_tools import *
+from module.Encoder import Deeplabv2
+from albumentations import Normalize
+from albumentations import *
+import ever as er
+
+def max_power_of_2(num : int):
+    p: int = 2
+    while p * 2 <= num:
+        p *= 2
+    return p
+
+
+class DcaAdapter(AdapterBase):
+    def __init__(self, size=1024, factor=2, mode="Urban"):
+        self.size = size
+        self.factor = factor
+        self.mode = mode
+
+    def set_size(self, size):
+        self.size = size
+
+    def process(self, image: np.ndarray):
+        # image - RGB
+        image = self._transform_image(image)
+        model = self._build_model()
+        raw_predictions = self._process(model, image)
+        predictions = self._postprocess_predictions(raw_predictions)
+        return predictions
+
+    # requires MxMx3 image
+    def _transform_image(self, image: np.ndarray):
+        image = Image.fromarray(image)
+        image = image.convert("RGB")
+        image = image.resize((self.size, self.size))
+        image = np.array(image)
+        print(image.shape)
+        mask = np.zeros([512, 512])
+        transformer = Compose([
+            Normalize(mean=(73.53223948, 80.01710095, 74.59297778),
+                    std=(41.5113661, 35.66528876, 33.75830885),
+                    max_pixel_value=1, always_apply=True),
+            # Normalize(mean=(123.675, 116.28, 103.53),
+            #           std=(58.395, 57.12, 57.375),
+            #           max_pixel_value=1, always_apply=True),
+            er.preprocess.albu.ToTensor()
+        ], is_check_shapes=False)
+        blob = transformer(image=image, mask=mask)
+        image = blob['image'].to(torch.device('cuda' if torch.cuda.is_available() else 'cpu'))
```

Comment:
What if I want to run on the CPU, although the GPU is available?

Reply:
Is it ok to fix it with using `os.environ["CUDA_VISIBLE_DEVICES"]` ?

Reply:
Nope

Reply:
It's better to explicitly give the user the ability to switch execution context